### PR TITLE
Fix Sqlite DB on Windows

### PR DESCRIPTION
**Pull request by rvl**
Relates to #703.
Based on the branch of #888.

# Overview

The wallet failed with:

    NUL: openFile: does not exist (No such file or directory)

This is due to runMigrationSilent which uses the silently library:

http://hackage.haskell.org/package/silently-1.2.5.1/docs/src/System.IO.Silently.html#line-27

To fix, we swap `runMigrationQuiet` for `runMigrationSilent` on Windows.

### CODE REVIEW

File: lib/core/src/Cardano/DB/Sqlite.hs [hunk 1]
```diff
@@ -74,13 +75,32 @@ import GHC.Generics
...
+-------------------------------------------------------------------------------}
+
+#if WINDOWS
+import Control.Monad.Trans.Reader
+    ( ReaderT )
+import Database.Persist.Sqlite
+    ( runMigration, showMigration )
+
+-- Replace the "runMigrationSilent" that uses "hSilence" which does not seem to
+-- work on Windows!
```
Comment (KtorZ) [inline lib/core/src/Cardano/DB/Sqlite.hs]: :man_facepalming: seriously

**Comment by rvl** [comment]
I was looking at fixing this in our fork of persistent and noticed that the recent persistent-2.10.2 release has a new function `runMigrationQuiet` that doesn't use `hSilence`. So I have updated the version and switched to that.

In `stack.yaml` we really shouldn't be using personal forks long term because it causes maintenance issues for the team.

It will be easier to keep persistent-* package versions synced if they are all supplied as `subdirs` of `extra-deps`.

**Comment by KtorZ** [comment]
@rvl 

> In stack.yaml we really shouldn't be using personal forks long term because it causes maintenance issues for the team.

I agree, this was only temporary until I could get proper access for our team. This has been updated and now points to:

https://github.com/input-output-hk/persistent/tree/cardano-wallet

**Review by KtorZ — APPROVED** [review]
LGTM (for the dev part)

**Comment by rvl** [comment]
Fixed base branch and cleared stack caches on Buildkite.

bors r+

**Comment by rvl** [comment]
The integration tests failed due to #931. Retrying.

bors r+

**Pull request merged**